refactor(curriculum): JavaScript date quiz (#58477)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-javascript-dates/66edd3711bb9f7efa73aef91.md

#### --text--

What is the corresponding month for `console.log(new Date(2003, 6, 27).getMonth());`?

#### --distractors--

> June

Reviewer: ```suggestion
June.
```

---

> April

Reviewer: ```suggestion
April.
```

---

> January

Reviewer: ```suggestion
January.
```

#### --answer--

> July

Reviewer: ```suggestion
July.
```

Reviewer 2: as mentioned in chat, we decided not to add periods for single words. Only full sentences 👍

#### --text--

What is the default locale used by the `toLocaleDateString()` method if no `locales` parameter is provided?

#### --distractors--

English (Great Britain).  

---

French (France).

---

English (United States).

#### --answer--

> The user's system locale.

Reviewer: I think we should have two similar options here. The answer stands out from the rest so someone who is unsure may just choose it. Maybe something like this:

```
`const d = new Date("2021-02-01");`
English (Great Britain).  

---

`const d = new Date("2003-06-06");`
The user's default system setting.

---

`const d = new Date("2001-04-08");`
English (United States).

#### --answer--

`const d = new Date("2025-01-01");`
The user's system locale.
```

Author: The options "The user's default system setting." and "The user's system locale." seem quite similar. I have updated the options as shown below:

```md

#### --distractors--

English (Great Britain).  

---

The locale closest to United States.

---

French (France).

#### --answer--

The user's system locale.

```

Let me know if this looks good.

### --question--

#### --text--

What gets assigned to `now` in the code below?

```js
const now = new Date();
```

#### --distractors--

The current time in milliseconds since January 1, 1990.

---

The current date in milliseconds minus the Unix epoch.

---

The current time in nanoseconds since January 1, 1990.

#### --answer--

> The current date and time based of your computer.

Reviewer 3: ```suggestion
The current date and time based on your computer system's clock.
```
